test/SlowTests/Issues/RavenDB-14880.cs

```diff
@@ -46,13 +46,14 @@ public async Task UpdateClientConfigurationOnlyWhenRequired(bool isServerWide)
                 WaitForIndexing(store);
 
                 var re = store.GetRequestExecutor(databaseName);
-                var configurationChanges = new HashSet<long>();
+                var configurationChanges = new List<long>();
 
                 re.ClientConfigurationChanged += (sender, tuple) =>
                 {
                     // since this is a cluster operation, when we fail-over to another node for reads,
                     // the other node might not get the configuration change yet and we might get the "old" configuration back
-                    configurationChanges.Add(re.ClientConfigurationEtag);
+                    if(configurationChanges.Contains(tuple.RaftCommandIndex) == false)
+                        configurationChanges.Add(tuple.RaftCommandIndex);
```

Comment:
Why using List instead of HashSet here?

Reply:
The values here are from `Hashing.Combine` so there is no meaning if a value is smaller or greater from another value.
The main problem here was that one node wasn't updated so while the client switch between the nodes because of the `ReadBalanceBehavior.RoundRobin` configuration the `ClientConfigurationChanged` is called multiple times.
This is the print from `HasClientConfigurationChanged`:
```
B index(0) != actual(1100310647005720893)
B index(1100310647005720893) != actual(1100310647005720893)
B index(1100310647005720893) != actual(1100310647005720893)
B index(1100310647005720893) != actual(1100310647005720893)
B index(1100310647005720893) != actual(-2545487602637056995)
B index(-2545487602637056995) != actual(-2545487602637056995)
C index(-2545487602637056995) != actual(-1212807392199088769)
B index(-1212807392199088769) != actual(-2545487602637056995)
C index(-2545487602637056995) != actual(-1212807392199088769)
B index(-1212807392199088769) != actual(-2545487602637056995)
C index(-2545487602637056995) != actual(-1212807392199088769)
B index(-1212807392199088769) != actual(-2545487602637056995)
```

I opened an issue to solve it - https://issues.hibernatingrhinos.com/issue/RavenDB-15548.

So to make a long story short I used `List` so the `Last` function will return the more updated value